### PR TITLE
feat: adding flags to enable dev mode; flags for custom path to localnet config

### PR DESCRIPTION
**Pull request by aorumbayev**
Fixes https://github.com/algorandfoundation/algokit-cli/issues/568

## Proposed Changes

  - Adds -dev/--no-dev flag for controlling devmode on algod container
  - Adds option to specify custom path to compose config directory

### CODE REVIEW

File: src/algokit/cli/localnet.py [hunk 1]
```diff
@@ -126,7 +128,26 @@ def config_command(*, engine: str | None, force: bool) -> None:
     "AlgoKit will not manage the configuration of named LocalNet instances, "
     "allowing developers to configure it in any way they need.",
 )
-def start_localnet(name: str | None) -> None:
+@click.option(
```
Comment (neilcampbell) [inline src/algokit/cli/localnet.py:131]: Do we want to ensure that a user doesn't supply both `name` and `config_path`?

File: src/algokit/core/sandbox.py [hunk 2]
```diff
@@ -46,9 +46,9 @@ def get_min_compose_version() -> str:
 
 
 class ComposeSandbox:
```
Comment (neilcampbell) [inline src/algokit/core/sandbox.py:48]: There is logic to detect if any configuration is out of date. Do we need to change that to ignore the value of dev mode?

File: src/algokit/core/sandbox.py [hunk 3]
```diff
@@ -184,6 +195,21 @@ def up(self) -> None:
         else:
             logger.warning("AlgoKit LocalNet failed to return a successful health check")
 
+    def recreate_container(self, container_name: str) -> None:
```
[the remark this answers is not in the document]
Comment (aorumbayev) [reply src/algokit/core/sandbox.py]: Restored to regular reset, recreate would result in faulty state

File: src/algokit/cli/localnet.py (hunk 1 above)
Comment (aorumbayev) [reply src/algokit/cli/localnet.py:132]: Why do you think doing both is not an option?

File: docs/features/localnet.md [hunk 4]
```diff
@@ -158,5 +187,6 @@ Running an interactive session ensures that you have control over the lifecycle
 - `--force`, `-f`: Force deletes stale codespaces and skips confirmation prompts. Defaults to explicitly prompting for confirmation.
 
 For more details about managing LocalNet in GitHub Codespaces, please refer to the [AlgoKit CLI reference documentation](../cli/index.md#codespace).
+U
```
Comment (neilcampbell) [inline docs/features/localnet.md:190]:
```suggestion
U
```

**Pull request merged**